universe/interface.go

```diff
@@ -1235,3 +1236,90 @@ type IgnoreTree interface {
 	QueryTuples(context.Context, asset.Specifier,
 		...IgnoreTuple) TupleQueryResp
 }
+
+// BurnLeaf is a type that represents a burn leaf within the universe tree.
+type BurnLeaf struct {
+	// UniverseKey is the key that the burn leaf is stored at.
+	UniverseKey LeafKey
+
+	// BurnProof is the burn proof that is stored within the burn leaf.
+	BurnProof *proof.Proof
+}
+
+// UniverseLeafNode returns the leaf node for the burn leaf.
+func (b *BurnLeaf) UniverseLeafNode() (*mssmt.LeafNode, error) {
+	var proofBuf bytes.Buffer
+	if err := b.BurnProof.Encode(&proofBuf); err != nil {
+		return nil, fmt.Errorf("unable to encode burn "+
+			"proof: %w", err)
+	}
+	rawProofBytes := proofBuf.Bytes()
+
+	return mssmt.NewLeafNode(rawProofBytes, b.BurnProof.Asset.Amount), nil
+}
+
+// AuthenticatedBurnLeaf is a type that represents a burn leaf within the
+// Universe tree. This includes the MS-SMT inclusion proofs.
+type AuthenticatedBurnLeaf struct {
+	*BurnLeaf
+
+	// BurnTreeRoot is the root of the ignore tree that the burn leaf
+	// resides within.
+	BurnTreeRoot mssmt.Node
+
+	// BurnProof is the universe inclusion proof for the burn leaf within
+	// the universe tree.
+	BurnProof *mssmt.Proof
+}
+
+// BurnDesc is a type that represents a burn leaf within the universe tree. This
+// is useful for querying the state without needing the proof itself.
+type BurnDesc struct {
+	// AssetSpec is the asset specifier for the burn leaf.
+	AssetSpec asset.Specifier
```

Comment:
Ah, I see we're using the specifier here. So perhaps the collision issue isn't present here, but still going through the diff...